src/test/suite/completion.test.ts

```diff
@@ -84,47 +89,107 @@ describe("Should do completion", () => {
     );
   });
   it("should prefer the popup when only popup is visible and there is no inline suggestion", async () => {
-    (await openADocWith("cons")).makeAChange("o");
+    await openADocWith("cons");
+
+    await makeAChange("o");
 
     await triggerPopupSuggestion();
 
     await triggerInline();
 
-    await triggerSelectionAppetence();
+    await emulationUserInteraction();
+
+    await triggerSelectionAcceptance();
 
     assertTextIsCommitted("console");
   });
   it("should prefer an inline when both popup and inline are visible", async () => {
-    (await openADocWith("cons")).makeAChange("o");
+    await openADocWith("cons");
+
+    await makeAChange("o");
 
     await triggerPopupSuggestion();
 
     mockInlineResponse();
 
     await triggerInline();
 
-    await triggerSelectionAppetence();
+    await triggerSelectionAcceptance();
 
     assertTextIsCommitted("console.log");
 
     await makeAndAssertFollowingChange();
   });
   it("should skip completion request on midline invalid position", async () => {
-    await openADocWith(" s");
+    await openADocWith("console s");
 
-    await moveToStartOfLinePosition();
+    await moveCursorToBeAfter("console");
 
-    mockInlineResponse();
+    await triggerInline();
+
+    await emulationUserInteraction();
+
+    verify(
+      stdinMock.write(new SimpleAutocompleteRequestMatcher(), "utf8")
+    ).never();
+  });
+  it("should request completion on midline valid position", async () => {
+    await openADocWith("consol })");
+
+    await moveCursorToBeAfter("consol");
```

Comment:
Not bad, now I do understand what it does 😀
I think it would be even better to give an absolute position, something like

```typescript 
moveCursorFromStart(6);
```

Reply:
But not critical - you can leave it this way as well